articles/azure/azs-how-to-retrieve-billing-api.md

```diff
@@ -0,0 +1,216 @@
+---
+title: How to retrieve Billing information from the CSP Portal via API
+description: How to retrieve Billing information from the CSP Portal via API
+services: azure-stack
+author: David Woffendin
+
+toc_rootlink: Operators
+toc_sub1: How To
+toc_sub2: 
+toc_sub3:
+toc_sub4:
+toc_title: Retrieve Billing information from the CSP Portal via API
+toc_fullpath: Operators/How To/azs-how-to-retrieve-billing-api.md
+toc_mdlink: azs-how-to-retrieve-billing-api.md
+---
+# Retrieving Billing information from the CSP Portal via API
+The purpose of this guide is to help obtain Billing information from the CSP Portal via the API.
+
+## Authentication 
+Authentication requires obtaining AAD token and then JWT token from the Partner Portal itself, appended to each request.
+
+**AAD Token**
+
+### Login endpoint
+```
+https://login.microsoftonline.com/{tenant id}/oauth2/token
+or
+https://login.windows.net/{tenant id}/oauth2/token
+```
+The tennit ID can either be:
+* Default value for tenant independent keys - common
+* The tennent ID, e.g. XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+* The tennet Domain e.g. example.your.domain.com
+
+### Body
+
+**grant_type**
```

Comment:
I would do these as a bulleted list, so for example, the first one would be:

- **grant_type** - This is the type of authentication scheme you will be using. In this example, the value is `password`.

Reply:
fixed